src/cli/rustup_mode.rs

```diff
@@ -824,28 +825,38 @@ async fn check_updates(cfg: &Cfg<'_>, opts: CheckOpts) -> Result<utils::ExitCode
                 let dist_version = distributable.show_dist_version().await?;
                 let mut update_a = false;
 
+                let styled_name = style(format!("{name} - ")).bold();
                 let message = match (current_version, dist_version) {
                     (None, None) => {
-                        format!("{name} - Cannot identify installed or update versions")
+                        let m = style("Cannot identify installed or update versions")
+                            .red()
+                            .bold();
+                        format!("{styled_name}{m}")
                     }
                     (Some(cv), None) => {
-                        format!("{name} - Up to date : {cv}")
+                        let m = style("Up to date").green().bold();
+                        format!("{styled_name}{m} : {cv}")
                     }
                     (Some(cv), Some(dv)) => {
+                        let m = style("Update available").yellow().bold();
                         update_a = true;
-                        format!("{name} - Update available : {cv} -> {dv}")
+                        format!("{styled_name}{m} : {cv} -> {dv}")
                     }
                     (None, Some(dv)) => {
+                        let m = style("Update available").yellow().bold();
                         update_a = true;
-                        format!("{name} - Update available : (Unknown version) -> {dv}")
+                        format!("{styled_name}{m} : (Unknown version) -> {dv}")
                     }
                 };
-                pb.set_style(ProgressStyle::with_template("{msg}").unwrap());
-                pb.finish_with_message(message.clone());
+                pb.set_style(ProgressStyle::with_template(message.as_str()).unwrap());
+                pb.finish();
                 Ok::<(bool, String), Error>((update_a, message))
             },
         );
 
+        // If we are running in a TTY, we can use `buffer_unordered` since
```

Comment:
I suspect this PR can be entirely or partially squashed back to one or more of the previous commits following the same principle... This comment for example feels a bit off.